R/linefuns.R

```diff
@@ -167,13 +167,70 @@ line_midpoint <- function(l, tolerance = NULL) {
 #' l <- routes_fast_sf[2:4, ]
 #' l_seg_multi = line_segment(l, segment_length = 1000)
 #' plot(sf::st_geometry(l_seg_multi), col = seq(nrow(l_seg_100)), lwd = 5)
-line_segment <- function(
+line_segment <- function(l, n_segments = NA, segment_length = NA) {
+  UseMethod("line_segment")
+}
+
+#' @export
+line_segment.sf <- function(
     l,
     n_segments = NA,
     segment_length = NA
     ) {
+
+  if (is.na(n_segments) && is.na(segment_length)) {
+    rlang::abort(
+      "`n_segment` or `segment_length` must be set.",
+      call = rlang::caller_env()
+    )
+  }
+
+  # if rsgeo is available use it
+  if (rlang::is_installed("rsgeo", version = "0.1.6")) {
+    # if CRS is NA then we can continue or if IsGeographic is NA
+    crs <- sf::st_crs(l)
+    is_geographic <- crs$IsGeographic
+
+    # if its NA set FALSE, if not keep
+    is_geographic <- ifelse(is.na(is_geographic), FALSE, is_geographic)
+
+    if (is.na(crs) || !is_geographic) {
+      # extract geometry and convert to rsgeo
+      geo <- rsgeo::as_rsgeo(sf::st_geometry(l))
+
+      # if n_segments is missing it needs to be calculated
+      if (is.na(n_segments)) {
+        l_length <- rsgeo::length_euclidean(geo)
+        n_segments <- max(round(l_length / segment_length), 1)
+      }
+
+      # segmentize the line strings
+      res <- rsgeo::line_segmentize(geo, n_segments)
+
+      # make them into sfc_LINESTRING
+      res <- sf::st_cast(sf::st_as_sfc(res), "LINESTRING")
```

Comment:
What is it without this casting process, a multilinestring?

Reply:
Correct